### ISSUE TITLE
Everything is as expected when batch size <= 64 but having nan loss after few epochs if batch size > 64

### ISSUE THREAD

**Issue by Tengger** [tensorflow/tensorflow#44026]
<em>When I was training my image dataset with InceptionResNetV2 everything works fine if the batch size is less than or equal to 64, however, if I set the batch size larger than 64 (For example, 128), after a few epochs the loss becomes nan</em>

**System information**
- OS Platform and Distribution: Linux-4.15.0-112-generic-x86_64-with-Ubuntu-18.04-bionic
- TensorFlow installed from (source or binary): NGC Container
- TensorFlow version (use command below): 2.2.0+nv
- Python version: 3.6.9
- Bazel version (if compiling from source): 2.0.0
- GCC/Compiler version (if compiling from source): c++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
- CUDA/cuDNN version: 11.0  
- GPU model and memory: Tesla V100 32510MiB

**Code**
##
##
    my_strategy = tf.distribute.MirroredStrategy()
    with my_strategy.scope():
        if args.model == 'usDL_InceptionResNetV2':
            print("train usDL_InceptionResNetV2...!")
            logging.info("train usDL_InceptionResNetV2 with {} GPUs...".format(args.gpus))
            model = models.usDL_InceptionResNetV2(args)

        elif args.model == 'usDL_InceptionV3':
            print("train usDL_InceptionV3...!")
            logging.info("train usDL_InceptionV3 with {} GPUs...".format(args.gpus))
            model = models.models.usDL_InceptionV3(args)

        else:
            print("No model is available!")
            sys.exit()

            # Replicates the model on multiple GPUs.
            #model = multi_gpu_model(model, gpus=args.gpus)

        ##
        ## compile the model
        ##

        if (args.model == "usDL_InceptionResNetV2"):

            # 2020-05-01
            logging.info("loss: categorical_crossentropy, optimizer: Adam()")
            model.compile(loss='categorical_crossentropy', optimizer=Adam(), metrics=['categorical_accuracy'])

            # model.compile(loss='binary_crossentropy', optimizer=Adam(), metrics=['acc'])

        else:
            logging.info("optimizer: Adam, loss: ssim_loss")
            model.compile(optimizer=Adam(), loss=ssim_loss, metrics=[ssim_loss, 'accuracy'])

        ##
        ## use call back functions
        ##

        ModelCheckpointFileName = os.path.join(model_save_dir, 'model_{epoch:02d}.h5')
        ckpt = ModelCheckpoint(ModelCheckpointFileName, monitor='val_loss', verbose=0, period=args.save_every)
        csvlogFileName = os.path.join(model_save_dir, 'log.csv')
        csv_logger = CSVLogger(csvlogFileName, append=True, separator=',')
        lr_sched = step_decay_schedule(initial_lr=args.lr)

    ##
    ## train
    ##
    ## The "steps_per_epoch" is not provoded here since the "__len__()" function
    ## has been provoded in the "My_Custom_Generator"
    ##

    print("train: start...")

    if (args.model == "usDL_InceptionResNetV2"):
        history = model.fit_generator(generator=train_generator,
                                      # steps_per_epoch=len(data)//args.batch_size,
                                      epochs=args.epoch,
                                      verbose=1,
                                      callbacks=[ckpt, csv_logger, lr_sched])

**Comment by gowthamkpr** [comment]
@Tengger Please give us the entire code to reproduce the issue. Thanks!
